internal/venice-common/src/main/java/com/linkedin/venice/init/ControllerClientBackedSystemSchemaInitializer.java

```diff
@@ -0,0 +1,147 @@
+      ControllerClient controllerClient,
+      VeniceSystemStoreType systemStore,
+      SchemaPresenceChecker schemaPresenceChecker,
+      boolean autoCreateWriteSystemStores,
+      UpdateStoreQueryParams updateStoreQueryParams,
+      boolean autoRegisterDerivedComputeSchema) {
+    super(systemStore, controllerClient.getClusterName(), updateStoreQueryParams, autoRegisterDerivedComputeSchema);
+    this.controllerClient = controllerClient;
+    this.systemStore = systemStore;
+    this.schemaPresenceChecker = schemaPresenceChecker;
+    this.autoCreateWriteSystemStores = autoCreateWriteSystemStores;
+  }
+
+  @Override
+  protected String getActualSystemStoreCluster() {
+    D2ServiceDiscoveryResponse response = controllerClient.discoverCluster(getSchemaStoreName());
+    if (response.isError()) {
+      if (STORE_NOT_FOUND.equals(response.getErrorType())) {
+        return null;
+      } else {
+        throw new VeniceException(response.getError());
+      }
+    }
+
+    return response.getCluster();
+  }
+
+  @Override
+  protected boolean doesStoreExist() {
+    StoreResponse response = controllerClient.retryableRequest(10, cc -> cc.getStore(getSchemaStoreName()));
+    if (response.isError()) {
+      if (STORE_NOT_FOUND.equals(response.getErrorType())) {
+        return false;
+      } else {
+        throw new VeniceException(response.getError());
+      }
+    }
+
+    return true;
+  }
+
+  @Override
+  protected boolean shouldCreateMissingSchemaStores() {
+    return autoCreateWriteSystemStores;
+  }
+
+  @Override
+  protected void createSchemaStore(Schema firstKeySchema, Schema firstValueSchema) {
+    NewStoreResponse newStoreResponse = controllerClient.createNewSystemStore(
+        systemStore.getZkSharedStoreName(),
+        VeniceConstants.SYSTEM_STORE_OWNER,
+        firstKeySchema.toString(),
+        firstValueSchema.toString());
+
+    if (newStoreResponse.isError()) {
+      throw new VeniceException(newStoreResponse.getError());
+    }
+  }
+
+  @Override
+  protected void updateStore(UpdateStoreQueryParams updateStoreQueryParams) {
+    controllerClient.updateStore(getSchemaStoreName(), updateStoreQueryParams);
+  }
+
+  @Override
+  protected SchemaEntry getRegisteredKeySchema() {
+    SchemaResponse response = controllerClient.getKeySchema(getSchemaStoreName());
+    return new SchemaEntry(response.getId(), AvroCompatibilityHelper.parse(response.getSchemaStr()));
+  }
+
+  @Override
+  protected Collection<SchemaEntry> getAllRegisteredValueSchemas() {
+    MultiSchemaResponse response = controllerClient.getAllValueSchema(getSchemaStoreName());
+
+    return Arrays.stream(response.getSchemas())
+        .map(schema -> new SchemaEntry(schema.getId(), AvroCompatibilityHelper.parse(schema.getSchemaStr())))
+        .collect(Collectors.toSet());
+  }
+
+  @Override
+  protected void registerValueSchema(Schema schema, int version) {
+    controllerClient.addValueSchema(getSchemaStoreName(), schema.toString(), version);
```

Comment:
Adding value schema with version/id is not supported in parent controller:
```
  /**
   * Unsupported operation in the parent controller.
   */
  @Override
  public SchemaEntry addValueSchema(
      String clusterName,
      String storeName,
      String valueSchemaStr,
      int schemaId,
      boolean doUpdateSupersetSchemaID) {
    throw new VeniceUnsupportedOperationException("addValueSchema");
  }
```

Reply:
Good catch. Will add this support

Reply:
Modified the code to communicate only with child controller. So, this is not needed now